src/libraries/Native/Unix/System.Globalization.Native/entrypoints.c

```diff
@@ -17,45 +18,65 @@ typedef uint16_t UChar;
-#define FCFuncEnd() (void*)0x01 /* FCFuncFlag_EndOfArray */ };
-
-#define QCFuncElement(name,impl) \
-    (void*)0x8 /* FCFuncFlag_QCall */, (void*)(impl), (void*)name,
-
-FCFuncStart(gPalGlobalizationNative)
-    QCFuncElement("ChangeCase", GlobalizationNative_ChangeCase)
-    QCFuncElement("ChangeCaseInvariant", GlobalizationNative_ChangeCaseInvariant)
-    QCFuncElement("ChangeCaseTurkish", GlobalizationNative_ChangeCaseTurkish)
-    QCFuncElement("CloseSortHandle", GlobalizationNative_CloseSortHandle)
-    QCFuncElement("CompareString", GlobalizationNative_CompareString)
-    QCFuncElement("EndsWith", GlobalizationNative_EndsWith)
-    QCFuncElement("EnumCalendarInfo", GlobalizationNative_EnumCalendarInfo)
-    QCFuncElement("GetCalendarInfo", GlobalizationNative_GetCalendarInfo)
-    QCFuncElement("GetCalendars", GlobalizationNative_GetCalendars)
-    QCFuncElement("GetDefaultLocaleName", GlobalizationNative_GetDefaultLocaleName)
-    QCFuncElement("GetICUVersion", GlobalizationNative_GetICUVersion)
-    QCFuncElement("GetJapaneseEraStartDate", GlobalizationNative_GetJapaneseEraStartDate)
-    QCFuncElement("GetLatestJapaneseEra", GlobalizationNative_GetLatestJapaneseEra)
-    QCFuncElement("GetLocaleInfoGroupingSizes", GlobalizationNative_GetLocaleInfoGroupingSizes)
-    QCFuncElement("GetLocaleInfoInt", GlobalizationNative_GetLocaleInfoInt)
-    QCFuncElement("GetLocaleInfoString", GlobalizationNative_GetLocaleInfoString)
-    QCFuncElement("GetLocaleName", GlobalizationNative_GetLocaleName)
-    QCFuncElement("GetLocales", GlobalizationNative_GetLocales)
-    QCFuncElement("GetLocaleTimeFormat", GlobalizationNative_GetLocaleTimeFormat)
-    QCFuncElement("GetSortHandle", GlobalizationNative_GetSortHandle)
-    QCFuncElement("GetSortKey", GlobalizationNative_GetSortKey)
-    QCFuncElement("GetSortVersion", GlobalizationNative_GetSortVersion)
-    QCFuncElement("GetTimeZoneDisplayName", GlobalizationNative_GetTimeZoneDisplayName)
-    QCFuncElement("IndexOf", GlobalizationNative_IndexOf)
-    QCFuncElement("InitICUFunctions", GlobalizationNative_InitICUFunctions)
-    QCFuncElement("InitOrdinalCasingPage", GlobalizationNative_InitOrdinalCasingPage)
-    QCFuncElement("IsNormalized", GlobalizationNative_IsNormalized)
-    QCFuncElement("IsPredefinedLocale", GlobalizationNative_IsPredefinedLocale)
-    QCFuncElement("LastIndexOf", GlobalizationNative_LastIndexOf)
-    QCFuncElement("LoadICU", GlobalizationNative_LoadICU)
-    QCFuncElement("NormalizeString", GlobalizationNative_NormalizeString)
-    QCFuncElement("StartsWith", GlobalizationNative_StartsWith)
-    QCFuncElement("ToAscii", GlobalizationNative_ToAscii)
-    QCFuncElement("ToUnicode", GlobalizationNative_ToUnicode)
-FCFuncEnd()
+#ifndef lengthof
+#define lengthof(rg)    (int)(sizeof(rg)/sizeof(rg[0]))
+#endif
+
+typedef struct
+{
+    const char* name;
+    const void* method;
+} Entry;
+
+static Entry s_globalizationNative[] =
```

Comment:
This isn't linker-friendly.  IT will keep every globalization function alive even if the IL Linker eliminates them all.

The data should be in a separate file from the lookup algorithm so that (for example) the WASM toolchain can supply a trimmed version of this table.

Reply:
This should work ok with the linker as long as you are careful to not reference this mapping table. I agree that it is fragile.

Consideration of linker friendliness was why these were in separate location (src/coreclr/src/libraries-native) originally.  #37670 folded the mapping table into the implementation.

Reply:
So, are we linker-friendly here or need changes?